File: lista-de-exercicios/exercicio-10/Matriz.test.ts
import { jest } from '@jest/globals';
import { Matriz } from './Matriz';

const matriz = new Matriz([[1, 2, 3], [4, 5, 6], [7, 8, 9]]);

describe('Classe Matriz', () => {
    test('Deve retornar a soma da diagonal principal de uma matriz quadrada', () => {
        expect(matriz.somar()).toBe(15);
    })
})

File: lista-de-exercicios-java/src/test/java/org/example/exercicio10/MatrizTest.java
package org.example.exercicio10;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class MatrizTest {

    Matriz matriz = new Matriz(new int[][] {
            {0, 1, 2, 3},
            {4, 5, 6, 7},
            {8, 9, 10, 11},
            {12, 13, 14, 15}
    });

    @DisplayName("Deve retornar a soma da diagonal principal de uma matriz quadrada")
    @Test
    void deveRetornarSomaDaDiagonalPrincipal() {
        assertEquals(30, matriz.somarDiagonal());
    }

    @DisplayName("Deve retornar a matriz e a soma da diagonal principal")
    @Test
    void deveRetornarStringResposta() {
        assertEquals("Dada a matriz: [[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11], [12, 13, 14, 15]]\n" +
                "A soma da diagonal principal é igual a: 30", matriz.resposta());
    }
}